lib/bibliothecary/analyser.rb

```diff
@@ -42,12 +42,40 @@ def map_dependencies(hash, key, type)
       end
 
       def analyse(folder_path, file_list)
-        file_list.map do |path|
+        analyses = file_list.map do |path|
           filename = path.gsub(folder_path, '').gsub(/^\//, '')
           next unless match?(filename)
           contents = File.open(path).read
           analyse_contents(filename, contents)
         end.compact
+
+        analyses.each do |analysis|
+          analysis[:related_paths] = []
```

Comment:
Is it worth trying to have this injected in `analyse_contents` instead of a second pass over the list?

Reply:
I did it that way at first. 
Conceptual problem: analyse_contents when called alone is then returning inaccurate information (empty array isn't a correct list of related paths)
Practical problem: all the expected results in spec/parsers/* break ;-)